workspace/apps/pidp/src/app/features/auth/pages/link-account-confirm/link-account-confirm.page.ts

```diff
@@ -61,11 +65,17 @@ export class LinkAccountConfirmPage implements OnInit {
     private linkAccountConfirmResource: LinkAccountConfirmResource,
     private router: Router,
     private loadingOverlayService: LoadingOverlayService,
+    private cookieService: CookieService,
+    private utilsService: UtilsService,
   ) {
     this.user$ = this.authorizedUserService.user$;
+    this.userName = '';
+    this.identityProvider = '';
   }
 
   public ngOnInit(): void {
+    this.userName = this.utilsService.decrypt(this.cookieService.get('UserName'));
```

Comment:
As mentioned in previous comment designing separate cookie service will separate the cookie logic.